Classes/Service/DemandService.php

```diff
@@ -64,7 +70,45 @@ public function getRestrictionsFromDemandArray(
         ExpressionBuilder $expressionBuilder
     ): CompositeExpression
     {
-        // TODO: Most of the code for getting restrictions in this class goes here
+        $demandArray = $this->restrictionsToInt($demandArray);
+        $expressions = [];
+
+        foreach ($demandArray as $key => $restrictions) {
```

Comment:
Don't forget to filter `$demandArray` so you only return restrictions for the supplied `$tables`. Check out `DemandArrayUtility::filterByTables()` https://github.com/pixelant/demander/blob/820b6ccdd05f558bab9b06b7c6388608ec497324/Classes/Utility/DemandArrayUtility.php#L61